internal/projector/slide/list_of_speakers.go

```
@@ -156,6 +175,30 @@ func renderListOfSpeakers(ctx context.Context, ds projector.Datastore, losFQID s
 	return b, fetch.Keys(), nil
 }
 
+// get_losID determines the losID and first current_projection of the reference_projector
+func getLosID(ctx context.Context, ContentObjectID string, fetch *datastore.Fetcher) (losID int, referenceProjectorID int, err error) {
+	parts := strings.Split(ContentObjectID, "/")
+	if len(parts) != 2 || parts[0] != "meeting" {
+		return losID, referenceProjectorID, fmt.Errorf("invalid Content_objectID %s. Expected a meeting-objectID", ContentObjectID)
+	}
+	meetingID, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return losID, referenceProjectorID, fmt.Errorf("invalid Content_objectID %s. Expected a numeric meeting_id", ContentObjectID)
+	}
+	referenceProjectorID = fetch.Int(ctx, "meeting/%d/reference_projector_id", meetingID)
+	referenceP7onIDs := fetch.Ints(ctx, "projector/%d/current_projection_ids", referenceProjectorID)
+
+	for _, pID := range referenceP7onIDs {
+		contentObjectID := fetch.String(ctx, "projection/%d/content_object_id", pID)
+		losID = fetch.Int(ctx, "%s/list_of_speakers_id", contentObjectID)
+
+		if losID != 0 {
+			break
```

What happens if all losIDs are 0? The function will return 0. Maybe you should either return an error or document this behavior in the docstring.

---

0 is a valid return value for losID and it is no error.